### DISCUSSION TITLE
Which files are required to create a delta update?

### DISCUSSION THREAD

**Discussion by AArnott** [Squirrel/Squirrel.Windows#1754] [Q&A] [answered]
Which files need to be downloaded from an S3 bucket to a build agent in order for `squirrel --releasify` to be able to properly create the delta package and RELEASES file? I'm *guessing* we need to download the RELEASES file and just one nupkg. I'm hoping we don't need to download the whole folder, since over time that could be huge.

Which nupkg should I download? The most recent? The newest (per semver)? The newest that's still older than the one being built (I'm thinking about servicing older versions)?

Has anyone an Azure Pipelines sample of how to do this efficiently?
I'm thinking about downloading the RELEASES file first then using that to discover the filename of the nupkg to then download before running `--releasify`.

**Comment by anaisbetts** [comment, marked as answer]
The SyncReleases tool does exactly what you need, it's designed to be run in CI and will do this all for you. It's only the latest full release (we don't build like roll-up deltas for 2-3 releases back)

The command you want to run right before you run `Squirrel --releasify` is:

```sh
./tools/squirrel.windows.2.0.1/tools/SyncReleases.exe -r ./path/to/future/releases/dir -u https://my-update-server.com
```

Also, remember that any place you can put an HTTPS URL in Squirrel you can test by putting a regular file path or a UNC path for internal distribution too

**Reply by AArnott** [reply]
Wonderful. Thank you.
